src/main/java/apoc/refactor/rename/Rename.java

```diff
@@ -61,27 +63,52 @@
 	 */
 	@Procedure(mode = Mode.WRITE)
 	@Description("apoc.refactor.rename.type(oldType, newType, [rels]) | rename all relationships with type 'oldType' to 'newType'. If 'rels' is provided renaming is applied to this set only")
-	public Stream<BatchAndTotalResultWithInfo> type(@Name("oldType") String oldType, @Name("newType") String newType, @Name(value = "rels", defaultValue = "[]") List<Relationship> rels) {
+	public Stream<BatchAndTotalResultWithInfo> type(@Name("oldType") String oldType,
+													@Name("newType") String newType,
+													@Name(value = "rels", defaultValue = "[]") List<Relationship> rels,
+													@Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
 		rels = rels.stream().map(r -> Util.rebind(tx, r)).collect(Collectors.toList());
-		List<Long> relIds = rels.stream().map(r -> r.getId()).collect(Collectors.toList());
 		String cypherIterate = rels != null && ! rels.isEmpty() ?
 				"UNWIND $rels AS oldRel WITH oldRel WHERE type(oldRel)=\""+oldType+"\" RETURN oldRel,startNode(oldRel) as a,endNode(oldRel) as b" :
 				"MATCH (a)-[oldRel:`"+oldType+"`]->(b) RETURN oldRel,a,b";
 		String cypherAction = "CREATE(a)-[newRel:`"+newType+"`]->(b)"+ "SET newRel+=oldRel DELETE oldRel";
-		Map<String, Object> parameters = MapUtil.map("batchSize", 100000, "parallel", true, "iterateList", true, "params", MapUtil.map("rels", rels));
+		final Map<String, Object> params = MapUtil.map("rels", rels);
+		Map<String, Object> parameters = getPeriodicConfig(config, params);
 		return getResultOfBatchAndTotalWithInfo(newPeriodic().iterate(cypherIterate, cypherAction, parameters), db, null, oldType, null);
 	}
 
+	@NotNull
```

Comment:
perhaps move to Periodic.java or Util.java?

Reply:
I prefer to leave it here because the defaults are custom defined in this procedure